ip-location-service.js

```diff
@@ -0,0 +1,55 @@
+"use strict";
+
+const path = require("path");
+const reader = require("@maxmind/geoip2-node").Reader;
+const AppConstants = require("./app-constants");
+
+let locationDb, timestamp;
+
+async function openLocationDb() {
+  if (locationDb && isFresh()) {
+    console.warn("Location database opened less than 24hrs: returning original instance.");
```

Comment:
Per chat with @groovecoder , it's understood that opening the db is expensive, but if we open it just once it runs the risk of stale data as long as the server is up, right?  So this was the easy compromise.  It would be good to know how often the MaxMind db actually changes, to get a better interval – I bet we could go 48hrs without reload.  In fact, I think I read from the MaxMind site that the db is updated once per week...  

Happy to hear alternate suggestions!

Reply:
Yeah, maybe I'm missing the core concept here&hellip;

Say we have two MaxMind databases:

```sh
GEOIP_GEOLITE2_CITY_FILENAME=GeoLite2-City-Test.mmdb
GEOIP_GEOLITE2_COUNTRY_FILENAME=GeoLite2-Country-Test.mmdb
```

And we want to keep those fresh. How are those being... loaded/deployed/updated?

I see below (L17) we're using `await reader.open(dbPath)`. But do those files get updated dynamically while the server is deployed somehow? Or is the data always the same during a deployment and only gets updated manually or via bumping some npm dependency or redeployment?
I guess my ignorant concern was that we have the two databases, and we do a production deployment and the .mmdb data never changes in the 1-4 weeks between deployments, but we still re-read the same files every 24h, loading the same data.
But if we aren't refreshing the data _dynamically_ somehow, I figured we just read the data once and don't bother with a freshness check.

Reply:
Yeah, this might be a good question for me to pose to SRE team since the database is managed within their k8 cluster.  Point taken that freshness might be overthinking this.

I confirmed that Geolite2 is updated once a week, but the paid GeoIP2 is updated twice weekly.  I will also check with SRE which they use.

Reply:
OK, I'll retract my initial concern. From the sounds of the Slack conversation, OPs is updating the .mmdb files 2x/week and updating the container's file system. So if we poll for freshness from our side once every 24 hours we know the data can never be older than 48-72h.

Reply:
@pdehaan I updated to reload the db upon client request every 3 days.  Let me know if you'd prefer reverting back to 24hrs 💪